dispatches/models/renewables_case/battery.py

```diff
@@ -0,0 +1,150 @@
+##############################################################################
+# DISPATCHES was produced under the DOE Design Integration and Synthesis
+# Platform to Advance Tightly Coupled Hybrid Energy Systems program (DISPATCHES),
+# and is copyright (c) 2021 by the software owners: The Regents of the University
+# of California, through Lawrence Berkeley National Laboratory, National
+# Technology & Engineering Solutions of Sandia, LLC, Alliance for Sustainable
+# Energy, LLC, Battelle Energy Alliance, LLC, University of Notre Dame du Lac, et
+# al. All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. Both files are also available online at the URL:
+# "https://github.com/gmlc-dispatches/dispatches".
+#
+##############################################################################
+# Import Pyomo libraries
+from pyomo.environ import Var, Param, NonNegativeReals, units as pyunits
+from pyomo.network import Port
+from pyomo.common.config import ConfigBlock, ConfigValue, In
+
+# Import IDAES cores
+from idaes.core import (Component,
+                        ControlVolume0DBlock,
+                        declare_process_block_class,
+                        EnergyBalanceType,
+                        MomentumBalanceType,
+                        MaterialBalanceType,
+                        UnitModelBlockData,
+                        useDefault)
+from idaes.core.util.config import list_of_floats
+import idaes.logger as idaeslog
+
+_log = idaeslog.getLogger(__name__)
+
+
+@declare_process_block_class("BatteryStorage", doc="Simple battery model")
+class BatteryStorageData(UnitModelBlockData):
+    """
+    Wind plant using turbine powercurve and resource data.
+    Unit model to convert wind resource into electricity.
+    """
+    CONFIG = ConfigBlock()
+    CONFIG.declare("dynamic", ConfigValue(
+        domain=In([False]),
+        default=False,
+        description="Dynamic model flag - must be False",
+        doc="""Battery does not support dynamic models, thus this must be False."""))
+    CONFIG.declare("has_holdup", ConfigValue(
+        default=False,
+        domain=In([False]),
+        description="Holdup construction flag",
+        doc="""Battery does not have defined volume, thus this must be False."""))
+
+    def build(self):
+        """Building model
+
+        Args:
+            None
+        Returns:
+            None
+        """
+        super().build()
+
+        self.dt = Param(within=NonNegativeReals,
```

Comment:
I would suggest more descriptive names for this, to help users in identifying what they mean/do.

Do we also need a comment describing why there is a separate `dt` from the `Flowsheet` time domain (and how the user should set this)?